schema.graphql

```diff
@@ -1,54 +1,58 @@
-type TokenStaked @entity {
+type Deposit @entity {
   id: ID!
   by: Account!
   amount: BigInt!
+  pool: Pool!
+  txHash: Bytes!
+  timestamp: BigInt!
 }
 
-type TokenUnstaked @entity {
+type UnstakedDeposit @entity {
   id: ID!
   by: Account!
   amount: BigInt!
+  pool: Pool!
+  txHash: Bytes!
+  timestamp: BigInt!
 }
 
-type TokenStakeLockUpdated @entity {
-  id: ID!
-  by: Account!
-  depositId: BigInt!
-  lockedFrom: BigInt!
-  lockedUntil: BigInt!
-}
-
-type StakeYieldClaimed @entity {
+type Reward @entity {
   id: ID!
-  by: Account!
+  for: Account!
   amount: BigInt!
+  pool: Pool!
+  txHash: Bytes!
+  timestamp: BigInt!
 }
 
-type TokenPoolWeightUpdated @entity {
+type Account @entity {
   id: ID!
-  by: Account!
-  fromVal: BigInt!
-  toVal: BigInt!
+  stakes: [Deposit!]! @derivedFrom(field: "by")
+  unstakes: [UnstakedDeposit!]! @derivedFrom(field: "by")
 }
 
-type Account @entity {
+type Pool @entity {
   id: ID!
-  deposits: [Deposit!]! @derivedFrom(field: "by")
-  stakes: [TokenStaked!]! @derivedFrom(field: "by")
-  unstakes: [TokenUnstaked!]! @derivedFrom(field: "by")
+  deposits: [Deposit!]! @derivedFrom(field: "pool")
 }
 
-type Deposit @entity {
+type TokenStakeLockUpdated @entity {
```

Comment:
Instead of having this as a separate entity, just add `lockedFrom` and `lockedUntil` on a deposit.